python/activator/middleware_interface.py

```diff
@@ -150,11 +155,14 @@ def _init_local_butler(self, butler: Butler):
         """
         # TODO: Replace registring the instrument with importing a "base" repo
         # structure from an export.yaml file.
-        self.instrument.register(butler.registry)
+        # TODO: Cannot do this until we have a way to only extract specific
+        # calibs, otherwise we get a unique constraint error on import.
+        # self.instrument.register(butler.registry)
```

Comment:
I don't see how calib selectivity is related to the uniqueness constraint (though I agree that not calling `instrument.register` is probably faster than debugging `saveDatasets(element=...)`).

Reply:
If we process the same detector in the same MiddlewareInterface, it will fail on `prep_butler` because all the calibs already exist, even independent of the `camera` dataset or the instrument registration.

Reply:
I still don't see why "extract specific calibs" matters; even with that capability, you'd still risk double-defining data IDs on later imports.

Reply:
If we can be more selective, we can just export the calibs we don't have, so we don't have the double-definition problem.

Reply:
Got it. Then I suggest rewording to something like "until we have a way to only extract calibs we don't already have". I thought you were talking about the validities.